doc/specifications/proposed/named_queries/developer_documentation.md

```diff
@@ -0,0 +1,202 @@
+    type: "article",
+    category: "development"
+```
+
+Each parameter will be matched exactly.
+The matcher will match if ALL of the provided parameters match what is contained in the hash.
+
+##### `QueryType\Expression`
+Uses Symfony's [Expression Language](http://symfony.com/en/doc/current/components/expression_language/index.html)
+for advanced matching.
+This matcher expects a valid expression language string as the input. The expression must be
+evaluated to a boolean. The parameters hash is available as `parameters`
+
+In the example below, 'type' must be either 'article' or 'blog_post'.
+
+```yaml
+match:
+  QueryType\Expression: "parameters['type'] in ['article', 'blog_post']"
+```
+
+### Query controller templates
+The templates used by this controller provide you with the results from the query. The results can be iterated over, and
+displayed using the object's properties or the `ez_content` controller actions.
+
+#### Available variables
+
+`location_list`/`content_list` | array | Array of search results, depending if the action (locationQuery or contentQuery)
+`list_count`                   | int   | Number of items in the resultset, within the limit if any
+`total_count`                  | int   | Total number of items in the search result
+
+## QueryType objects
+To make a new QueryType available to the Query Controller, you need to create a class that implements the QueryType
+interface, and register it as such in the service container.
+
+The interface has two methods:
+- `getName(): string`
+  Returns the QueryType name.
+- `getQuery( array $parameters = null ): Query`
+  It returns a Query object, that can be either a LocationQuery or a (Content) Query.
+  It accepts an optional hash of parameters that can be used in any way in the QueryType's implementation
+
+### Parameters
+A QueryType may accept parameters, depending on the implementation. They can be used in any way, such as:
+  - customizing an element's value (limit, ContentType identifier, etc)
+  - conditionally adding/removing criteria from the query
+  - setting the limit/offset
+
+The implementations should use Symfony's `OptionsResolver` for parameters handling and resolution.
+
+### QueryType example: latest content
+This QueryType returns a LocationQuery that searches for the 10 last published content, order by reverse
+publishing date. It accepts an optional `type` parameter, that can be set to a ContentType identifier.
+identifier:
+
+```php
+<?php
+namespace AcmeBundle\Ez\QueryType;
+
+use eZ\Publish\Core\MVC\Query\QueryType;
+use eZ\Publish\API\Repository\Values\Content\Query;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+
+class LatestContentQueryType implements QueryType
+{
+    public function getQuery(array $parameters = [])
+    {
+        $criteria[] = new Criterion\Visibility(Criterion\Visibility::VISIBLE);
+        if (isset($parameters['type'])) {
+            $criteria[] = new Criterion\ContentTypeIdentifier('article');
```

Comment:
It wasn't part of the initial example, but I could add it yes. Maybe it's a good opportunity to actually add the ~holy grail~ OptionsResolver ;-)